### PR TITLE
[Do not MERGE] Enable test_data_parallel

### PR DESCRIPTION
**Pull request by jaglinux**
Use the earlier way of repeat_test instead of instantiate_device_type_tests()

ToDo: debug why the newer method of test fails

### CODE REVIEW

File: test/distributed/test_data_parallel.py [hunk 1]
```diff
@@ -881,7 +877,7 @@ def forward(self, input):
         self.assertEqual(out, expected_out, atol=dtype2prec_DONTUSE[dtype], rtol=0)
 
 
-instantiate_device_type_tests(TestDataParallelDeviceType, globals())
```
Comment (pruthvistony) [inline test/distributed/test_data_parallel.py:884]: Unwanted change.

**Review by pruthvistony — COMMENTED** [review]
Not sure of making this change, it will mask a ROCm issue. Changing the have test cases are executed may be changing the runtime behavior (timing).

File: test/distributed/test_data_parallel.py (hunk 1 above)
Comment (jaglinux) [reply test/distributed/test_data_parallel.py:884]: That change is making the test fail.

**Comment by jaglinux** [comment]
> Not sure of making this change, it will mask a ROCm issue. Changing the have test cases are executed may be changing the runtime behavior (timing).

I am not recommending to merge this. Pointing out why the tests started failing.
Either the introduction of instantiate_device_type_tests() has exposed the bug in ROCm or there might be issue in the test setup. That is yet to be determined.
The test case has not been changed, only the setup and execution. The test was running fine earlier and even now if we use the old way of testing.

**Comment by jaglinux** [comment]
Updated with [Do not MERGE] . Intention is to debug further with the new findings.

File: test/distributed/test_data_parallel.py (hunk 1 above)
Comment (pruthvistony) [reply test/distributed/test_data_parallel.py:884]: Oops, wrong comment. We are removing device side instants of the test cases.

**Comment by jithunnair-amd** [comment]
retest pytorch please

**Pull request closed without merging**